services/state/blockrecords/block_info.proto

```diff
@@ -39,12 +39,12 @@ message BlockInfo {
   /**
    * The consensus time of the first transaction of the last block, this is the last completed immutable block.
    */
-  Timestamp first_cons_time_of_last_block = 2;
+  Timestamp first_cons_time_of_last_block = 3;
   /**
    * SHA384 48 byte hashes of the last 256 blocks in single byte array.
    * First 48 bytes is the oldest block.
    * Last 48 bytes is the newest block, which is the last fully completed immutable block.
    * If we are shortly after genesis and there are less than 256 blocks then this could contain less than 256 hashes.
    */
-  bytes block_hashes = 3;
+  bytes block_hashes = 2;
```

Comment:
When reordering, while not _necessary_, it is helpful to actually move the fields within the file.

Reply:
That's reasonable.  It's not necessary, but it still may be helpful to follow up with a reorder PR after this, just to make working with these easier.

Reply:
Speaking of which, what is the criteria for deciding what order to put the fields in?

Reply:
Ordering methodology is explained in the issue associated with this PR. https://github.com/hashgraph/hedera-services/issues/7684

Reply:
Overall I'm skeptical of the reasoning. I like that we're thinking about how to best order things. In the end it won't matter that much because when we add new fields (which we will no doubt do), they will completely break this ordering. We cannot change the field orders without breaking compatibility. So we will add new booleans at the end, for example, which breaks the reason for reordering fields in the first place.

Maybe it makes sense to order based on what is most likely to be set. Future fields will most likely be left at their default value (since there will be many existing objects with no knowledge of it, so the default will be quite prevalent).

Reply:
We could leave gaps to be filled in by possible future values, i.e. reserve 1-8 for booleans, etc.

Reply:
Later adds will be less efficient, certainly, but starting from a well packed structure saves space regardless, as that portion of the structure on disk will still be efficient, even if we add less efficiently packed elements later.
For very small object it's probably not a big difference, but for things like NFTs and accounts it's enough that it is worth making things as efficiently stored as we reasonably can.

Reply:
Done